### PR TITLE
Initial Commit for Project Unity

### PR DESCRIPTION
**Pull request by sanjayms01**

### CODE REVIEW

**Review by aishwaryamathuria — APPROVED** [review]

File: unitylibs/core/steps/app-connector.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+async function continueInApp(cfg, appName, btnConfig) {
+  const { unityEl, unityWidget, connectorApiEndPoint, apiKey, errorToastEvent } = cfg;
+  const continuebtn = unityWidget.querySelector(`continue-in-${appName}`);
+  if (continuebtn) return continuebtn;
+  const btn = await createActionBtn(btnConfig, `continue-in-app continue-in-${appName}`, true, true);
+  btn.addEventListener('click', async () => {
+    const data = getPreludeData(cfg);
+    const connectorOptions = {
+      method: 'POST',
+      headers: {
```
Comment (amitbikram) [inline unitylibs/core/steps/app-connector.js:38]: Can we move header to a separate function in utils. I am seeing it is being used at multiple places.

File: unitylibs/core/features/progress-circle/progress-circle.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { createTag } from '../../../scripts/utils.js';
+
+export default function createProgressCircle(cfg) {
+  const { unityEl, targetEl, progressCircleEvent } = cfg;
+  const pdom = `<div class="spectrum-ProgressCircle-track"></div>
+  <div class="spectrum-ProgressCircle-fills">
```
Comment (amitbikram) [inline unitylibs/core/features/progress-circle/progress-circle.js:6]: Can we make it a constant outside the function

File: unitylibs/core/steps/upload-btn.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+function showErrorToast(cfg, className) {
+  const { unityEl } = cfg;
+  const msg = unityEl.querySelector(className)?.nextSibling.textContent;
+  document.querySelector('.unity-enabled .interactive-area .alert-holder .alert-toast .alert-text p').innerText = msg;
+  document.querySelector('.unity-enabled .interactive-area .alert-holder').style.display = 'flex';
+}
+
+export function createErrorToast(cfg) {
+  const alertImg = `<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (amitbikram) [inline unitylibs/core/steps/upload-btn.js:11]: Is it possible to move it separate SVG file inside code and load on demand?

File: unitylibs/core/workflow/workflow.js [hunk 4]
```diff
@@ -1,3 +1,138 @@
...
+import {
+  createTag,
+  loadStyle,
+  setUnityLibs,
+  getUnityLibs,
+  unityConfig,
+  defineDeviceByScreenSize,
+  getConfig
+} from '../../scripts/utils.js';
+import { createErrorToast } from '../steps/upload-btn.js';
```
Comment (amitbikram) [inline unitylibs/core/workflow/workflow.js:10]: I think we do not need this on page load please correct if I am missing something.  Can we load it on demand

File: unitylibs/core/workflow/workflow.js [hunk 5]
```diff
@@ -1,3 +1,138 @@
...
+import createProgressCircle from '../features/progress-circle/progress-circle.js';
```
Comment (amitbikram) [inline unitylibs/core/workflow/workflow.js:11]: Same as above.

**Review by amitbikram — CHANGES_REQUESTED** [review]

File: unitylibs/core/features/progress-circle/progress-circle.js (hunk 2 above)
Comment (sanjayms01) [reply unitylibs/core/features/progress-circle/progress-circle.js:4]: Fixed

File: unitylibs/core/workflow/workflow-photoshop/workflow-photoshop.js [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+  getHeaders,
+  loadImg,
+  createActionBtn,
+  loadSvg,
+  decorateDefaultLinkAnalytics,
+  createIntersectionObserver,
+} from '../../../scripts/utils.js';
+import { uploadAsset } from '../../steps/upload-step.js';
+import initAppConnector from '../../steps/app-connector.js';
+import createUpload from '../../steps/upload-btn.js';
```
Comment (amitbikram) [inline unitylibs/core/workflow/workflow-photoshop/workflow-photoshop.js:13]: Can we try to import this and initialize when on demand like we did for toast

File: unitylibs/scripts/utils.js [hunk 7]
```diff
@@ -14,29 +14,167 @@ export const [setLibs, getLibs] = (() => {
...
+    if (swapOrder) actionBtn.prepend(btnTxt);
+    else actionBtn.append(btnTxt);
+  }
+  return actionBtn;
 }
 
-export async function useMiloSample() {
-  const { createTag } = await import(`${getLibs()}/utils/utils.js`);
+async function createErrorToast() {
+  const alertImg = await fetch(`${getUnityLibs()}/img/icons/alert.svg`).then((res) => res.text());
```
Comment (amitbikram) [inline unitylibs/scripts/utils.js:108]: Both the SVGs can be loaded in parallel.

**Review by amitbikram — APPROVED** [review]
@sanjayms01 This looks good. There is few minor comments we can take up in future PRs.

File: unitylibs/core/steps/upload-btn.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  const a = await createActionBtn(li, 'show');
+  const input = createTag('input', { class: 'file-upload', type: 'file', accept: 'image/png,image/jpg,image/jpeg', tabIndex: -1 });
+  a.append(input);
+  a.addEventListener('keydown', (e) => {
+    if (e.key === 'Enter') input.click();
+  });
+  a.addEventListener('change', async (e) => {
+    const file = e.target.files[0];
+    if (!file) return;
+    if (file.size > 400000000) {
```
[the remark this answers is not in the document]
Comment (sanjayms01) [reply unitylibs/core/steps/upload-btn.js]: Fixed

File: unitylibs/core/workflow/workflow-photoshop/workflow-photoshop.js (hunk 6 above)
Comment (sanjayms01) [reply unitylibs/core/workflow/workflow-photoshop/workflow-photoshop.js]: Fixed

File: unitylibs/scripts/utils.js (hunk 7 above)
Comment (sanjayms01) [reply unitylibs/scripts/utils.js]: Fixed

**Pull request merged**